### ISSUE TITLE
Docker Compose + parallel updates results in creation of multiple networks

### ISSUE THREAD

**Issue by maiamcc** [windmilleng/tilt#2817]
Reported by Anthony Kroh and by @tmc: after `tilt up`, network list looks something like:
```
docker network list
NETWORK ID          NAME                DRIVER              SCOPE
1359b387b864        bridge              bridge              local
658473c355d8        docker_default      bridge              local
dad8a3d57d87        docker_default      bridge              local
98847634f0da        docker_default      bridge              local
080784b486b2        host                host                local
66f68204ad20        none                null                local
```
and containers don’t come up with the error: `3 matches found based on name: network docker_default is ambiguous`

Because for each DC resource we call docker compose up ____ which presumably tries to create the network if it doesn’t exist, so doing that 3x in parallel results in… 3x the network being created.

Workarounds:
1. turn off parallel updates (`update_settings(max_parallel_updates=1`) )
2. create the network FIRST with a local_resource call (https://docs.docker.com/engine/reference/commandline/network_create/), and specify that local resource [as a dependency](https://docs.tilt.dev/resource_dependencies.html) of each other docker compose resource (so that Tilt won’t run docker compose up X until it’s run docker network create Y)

However, we should support Docker Compose + parallel updates out of the box -- maybe Tilt should do the work of pre-creating the network? Some other solution?

**Comment by nicks** [comment]
I feel like this one is an interesting mirror of https://github.com/windmilleng/tilt/issues/2770. We want the image builds to happen in parallel, but the deploys (e.g. creation of networks) to happen in serial

**Comment by nicks** [comment]
this should be fixed in tilt v0.12.1
